src/main/java/de/tum/in/www1/artemis/repository/metis/conversation/OneToOneChatRepository.java

```diff
@@ -25,24 +23,27 @@ public interface OneToOneChatRepository extends JpaRepository<OneToOneChat, Long
 
     Set<OneToOneChat> findAllByConversationParticipantsContaining(ConversationParticipant conversationParticipant);
 
-    @EntityGraph(type = LOAD, attributePaths = { "conversationParticipants.user.groups" })
     @Query("""
             SELECT DISTINCT oneToOneChat
             FROM OneToOneChat oneToOneChat
-                LEFT JOIN oneToOneChat.conversationParticipants conversationParticipant
-                LEFT JOIN FETCH oneToOneChat.conversationParticipants
+                LEFT JOIN FETCH oneToOneChat.conversationParticipants conversationParticipant
+                LEFT JOIN FETCH conversationParticipant.user user
+                LEFT JOIN FETCH user.groups
             WHERE oneToOneChat.course.id = :courseId
                 AND (oneToOneChat.lastMessageDate IS NOT NULL OR oneToOneChat.creator.id = :userId)
-                AND conversationParticipant.user.id = :userId
+                AND user.id = :userId
             ORDER BY oneToOneChat.lastMessageDate DESC
             """)
     List<OneToOneChat> findActiveOneToOneChatsOfUserWithParticipantsAndUserGroups(@Param("courseId") Long courseId, @Param("userId") Long userId);
 
-    @EntityGraph(type = LOAD, attributePaths = { "conversationParticipants.user.groups" })
     @Query("""
             SELECT o FROM OneToOneChat o
                 LEFT JOIN FETCH o.conversationParticipants p1
+                LEFT JOIN FETCH p1.user u1
+                LEFT JOIN FETCH u1.groups
                 LEFT JOIN FETCH o.conversationParticipants p2
+                LEFT JOIN FETCH p2.user u2
+                LEFT JOIN FETCH u2.groups
             WHERE o.course.id = :courseId
```

Comment:
This looks quite strange, why do we use `o.conversationParticipants` twice here?

Reply:
I didn't change anyting there functionality-wise. But I think the new implementation should do the same trick. Please verify.